### PR TITLE
Do less work to add/remove workers to/from PartialRange

### PR DESCRIPTION
**Pull request by jcorbin**
r @Raynos @rf

### CODE REVIEW

**Comment by jcorbin** [comment]
This gets the torch difference between full and partial refresh peers to within a fraction of a percent.

File: service-proxy.js [hunk 1]
```diff
@@ -1312,7 +1308,7 @@ function pruneSinglePeer(hostPort, pruneInfo) {
 };
 
 ServiceDispatchHandler.prototype.reapSinglePeer =
-function reapSinglePeer(hostPort, serviceNames) {
+function reapSinglePeer(hostPort, serviceNames, now) {
```
Comment (rf) [inline service-proxy.js]: `now` will be undefined, no? is it needed?

Comment (jcorbin) [inline service-proxy.js]: good catch, fixed

File: partial_range.js [hunk 2]
```diff
@@ -81,6 +81,28 @@ function compute(relays, workers, now) {
...
 };
 
+PartialRange.prototype.addWorker =
+function addWorker(hostPort, now) {
+    var i = sortedIndexOf(this.workers, hostPort);
+    if (i >= 0) {
+        return;
+    }
+
+    this.workers.splice(~i, 0, hostPort);
```
Comment (Raynos) [inline partial_range.js]: y u do dis.

**Comment by Raynos** [comment]
Seems legit.

**Pull request merged**